### DISCUSSION TITLE
Adding Bootstrap5 and SASS

### DISCUSSION THREAD

**Discussion by OxyTJ** [vercel/turborepo#2304] [Help] [answered]
I have a `vite` app and a shared `lib` package, both of which need access to Bootstrap5 and SASS. When installing bootstrap to either workspace, it's still installing in the root `node_modules` directory rather than within the workspace, so I'm unable to import with `import 'bootstrap/dist/css/bootstrap.min.css'` and `import 'bootstrap/dist/js/bootstrap.min.js'`.

npm commands I'm using to install:
```
npm i -w=client-app bootstrap @popperjs/core
npm i -D -w=client-app sass sass-resources-loader

npm i -w=lib bootstrap @popperjs/core
npm i -D -w=lib sass sass-resources-loader
```

I noticed I had a similar problem with a different application when using TailwindCSS, Flowbite and Flowbite-React - Modules being installed in root, out of context of each workspace.

Am I missing something?

[Turborepo GitHub Discussion](https://github.com/vercel/turbo/discussions/2304)

[StackOverflow Post](https://stackoverflow.com/questions/74187310/turborepo-adding-bootstrap5-and-sass)

[Reddit Post](https://www.reddit.com/r/vitejs/comments/yco6vw/turborepo_adding_bootstrap5_and_sass/)

**Comment by CodeShakingSheep** [comment]
@OxyTJ How were you able to solve this? I'm running in the same issue when I want to import bootstrap CSS.

**Reply by OxyTJ** [reply]
@CodeShakingSheep wish I remembered specifics, but been too long and not seeing anything different than what I have above when I do `history | grep npm i`. However, looks like I'm importing bootstrap like the following:

```
@import 'bootstrap/scss/bootstrap';
```

**Reply by CodeShakingSheep** [reply, marked as answer]
I figured out a way to do it inside my vite app. In `vite.config.ts` I defined an alias like this:

```ts
// https://vitejs.dev/config/
export default defineConfig({
  [...]
  resolve: {
    alias: {
      '~bootstrap': path.resolve(__dirname, '../../node_modules/bootstrap')
    }
  }
});
```
Note the `../../` part in the path to tell the app to look in the root `node_modules` directory for bootstrap. Then I import it in my .scss file like this 
```scss
@import "~bootstrap/scss/bootstrap";
```
It works like this. However, I'm curious if there is a way to tell turbo to install the modules in the project specific `node_modules` directory instead of the root `node_modules` directory. Can anybody comment on that?

**Reply by anthonyshew** [reply]
It's the responsibility of the package manager to coordinate packages (e.g. assign them a place on disk, resolve them for your application). Depending on your package manager, your configuration, and the dependencies you are installing, you may find that a dependency is installed either in a package or at the root of the workspace.

Your answer is correct for telling Vite to resolve a dependency deeper in your file tree!
